### PR TITLE
Change remaining API calls to use websockets

### LINKED ISSUE #130: Use WebSocket channel for API requests
The log events are currently sent to ZAP with HTTP, which means more resources are used than if sent through the WebSocket channel. Is there any reason not to use that instead?

Sample HTTP request:
```
GET https://zap//zapCallBackUrl/-4262183702902603983/hud/action/log/?record=00:00:00%20GMT+0000%20(WEST)%20INFO%20heartbeat:%20heartbeat HTTP/1.1
Host: zap
User-Agent: Mozilla/5.0 (X11; Linux x86_64; rv:59.0) Gecko/20100101 Firefox/59.0
Accept: */*
Accept-Language: en-GB,en;q=0.5
Accept-Encoding: gzip, deflate, br
Referer: https://zap//zapCallBackUrl/-3596693309352194563?name=management.html
origin: https://zap
Connection: keep-alive
```

### PR DESCRIPTION
**Pull request by psiinon**
Fixes #130

### CODE REVIEW

**Comment by psiinon** [comment]
This pull request **introduces 1 alert** when merging 39be64a0f419a070468b9d55581ee342853c8c57 into 8df623ad006a0fafe516e0a331bb08f97c857957 - [view on LGTM.com](https://lgtm.com/projects/g/zaproxy/zap-hud/rev/pr-9ee7244ca6d8b89a4a5e6689d2582c830f9d12c4)

**new alerts:**

* 1 for Superfluous trailing arguments

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by psiinon** [comment]
This pull request **introduces 1 alert** when merging 42d58dbb22523238850ce418c8ddca89908cc993 into 8df623ad006a0fafe516e0a331bb08f97c857957 - [view on LGTM.com](https://lgtm.com/projects/g/zaproxy/zap-hud/rev/pr-303a10b2d8bc3be123b54ec841925c21f8ac7e2a)

**new alerts:**

* 1 for Superfluous trailing arguments

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by psiinon** [comment]
This pull request **introduces 2 alerts** when merging 645fb2105f85674b75e2c3e3e013311e9ae52259 into 8df623ad006a0fafe516e0a331bb08f97c857957 - [view on LGTM.com](https://lgtm.com/projects/g/zaproxy/zap-hud/rev/pr-941ca21a55ac18aaf356598282b8aba1fe74fcc2)

**new alerts:**

* 1 for Superfluous trailing arguments
* 1 for Unused variable, import, function or class

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by psiinon** [comment]
The main work is all done now, just needs some tidying up and more testing...

**Comment by psiinon** [comment]
This pull request **introduces 2 alerts** when merging f5a1e8257f8a6b9da0312bae4f83eeace2af8c58 into 8df623ad006a0fafe516e0a331bb08f97c857957 - [view on LGTM.com](https://lgtm.com/projects/g/zaproxy/zap-hud/rev/pr-8d65209605eadb4fd8c1eb800b5d18f3c421ae61)

**new alerts:**

* 1 for Superfluous trailing arguments
* 1 for Unused variable, import, function or class

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by psiinon** [comment]
Hopefully fixed lgtm warnings...

**Comment by psiinon** [comment]
Oh, this does need this websocket PR in order to work: https://github.com/zaproxy/zap-extensions/pull/1924 :)
This is required for the HTML report. And this (recently merged) PR https://github.com/zaproxy/zap-extensions/pull/1921 is required for break to work.

File: src/test/java/org/zaproxy/zap/extension/hud/ui/firefox/badsite/BadSiteUnitTest.java [hunk 1]
```diff
@@ -167,30 +141,4 @@ private String getHudUtilsFilesUrl(HUD hud) throws MalformedURLException {
...
-     * @throws MalformedURLException
-     */
-    private String getHudApiUrl(HUD hud) throws MalformedURLException {
-
-        // Its worth noting that target sites wont be able to do this ;)
-        // We're just checking that even if the API URL is leaked somehow that the target site wont
-        // be able to use it
-        hud.getWebDriver().get(getHudUtilsFilesUrl(hud));
-        String utilsSrc = hud.getWebDriver().getPageSource();
-        int defnOffset = utilsSrc.indexOf("ZAP_HUD_API");
```
Comment (thc202) [inline src/test/java/org/zaproxy/zap/extension/hud/ui/firefox/badsite/BadSiteUnitTest.java]: There are still some references to `ZAP_HUD_API` in this test class that should be removed.

**Comment by thc202** [comment]
Working fine here, didn't notice any performance issue.

File: src/test/java/org/zaproxy/zap/extension/hud/ui/firefox/badsite/BadSiteUnitTest.java (hunk 1 above)
Comment (psiinon) [reply src/test/java/org/zaproxy/zap/extension/hud/ui/firefox/badsite/BadSiteUnitTest.java]: Fixed

**Comment by psiinon** [comment]
This pull request **introduces 1 alert** when merging 1f3643a00916326c7f252dc2ea9db789f5b447c9 into 8df623ad006a0fafe516e0a331bb08f97c857957 - [view on LGTM.com](https://lgtm.com/projects/g/zaproxy/zap-hud/rev/pr-868c80608beb4f4d90264364b038a5d39f1dd488)

**new alerts:**

* 1 for Unused variable, import, function or class

---

*Comment posted by [LGTM.com](https://lgtm.com)*

File: src/test/java/org/zaproxy/zap/extension/hud/ui/firefox/badsite/BadSiteUnitTest.java [hunk 2]
```diff
@@ -78,67 +74,36 @@ public void checkServerTrusted(
...
-            // This class should not expect a IOException in case this is thrown above
-            fail();
-        } catch (IOException e) {
-            // Expected
-        }
-    }
-
     @Test
-    public void cannotAccessApiKeyWhenLoadingUtilsAsScript(FirefoxDriver driver)
+    public void cannotAccessWebSocketsUrlWhenLoadingUtilsAsScript(FirefoxDriver driver)
```
Comment (thc202) [inline src/test/java/org/zaproxy/zap/extension/hud/ui/firefox/badsite/BadSiteUnitTest.java]: The `Utils` needs to be updated.

File: src/test/java/org/zaproxy/zap/extension/hud/ui/firefox/badsite/BadSiteUnitTest.java [hunk 3]
```diff
@@ -78,67 +74,36 @@ public void checkServerTrusted(
...
 
-        // LOG_ERROR is a public var - this should succeed
-        driver.executeScript("return LOG_ERROR;");
-
         // Now try to make use of it
+
+        Object result = driver.executeScript("return ZAP_HUD_FILES;");
+        assertTrue(result.toString().startsWith("https://zap//zapCallBackUrl"));
         try {
             // ZAP_HUD_API should be protected via a closure
```
Comment (thc202) [inline src/test/java/org/zaproxy/zap/extension/hud/ui/firefox/badsite/BadSiteUnitTest.java]: Needs update.

**Comment by psiinon** [comment]
Hopefully all comments addressed :)

**Comment by thc202** [comment]
Thanks!

**Comment by psiinon** [comment]
@dscrobonia would be great to get your feedback on this :)

**Pull request merged**